File: regexlib/python_re_test_file/regexlib_1626.py
# 1626
# ([A-Z]:\\[^/:\*\?<>\|]+\.\w{2,6})|(\\{2}[^/:\*\?<>\|]+\.\w{2,6})
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:""+"\\\\"*5000+"@1 _SLQ_1"

import re
from time import perf_counter

regex = """([A-Z]:\\[^/:\*\?<>\|]+\.\w{2,6})|(\\{2}[^/:\*\?<>\|]+\.\w{2,6})"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "\\\\" * i * 10000 + "@1 _SLQ_1"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")

File: regexlib/PoCs/Regexlib_Java_8/src/regexlib_1626.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class regexlib_1626 {
    /* 1626
     * ([A-Z]:\\[^/:\*\?<>\|]+\.\w{2,6})|(\\{2}[^/:\*\?<>\|]+\.\w{2,6})
     * POLYNOMIAL
     * nums:2
     * POLYNOMIAL AttackString:""+"\\\\"*5000+"@1 _SLQ_1"
     */
    public static void main(String[] args) throws InterruptedException {
        String regex = "([A-Z]:\\\\[^/:\\*\\?<>\\|]+\\.\\w{2,6})|(\\\\{2}[^/:\\*\\?<>\\|]+\\.\\w{2,6})";
        for (int i = 0; i < 1000; i++) {
            StringBuilder attackString = new StringBuilder();
            // 前缀
            attackString.append("");
            // 歧义点
            for (int j = 0; j < i * 10000; j++) {
                attackString.append("\\\\");
            }
            // 后缀
            attackString.append("@1 _SLQ_1");
//            System.out.println(attackString);
            long time1 = System.nanoTime();
//            boolean isMatch = Pattern.matches(regex, attackString);
            boolean isMatch = Pattern.compile(regex).matcher(attackString).find();
            long time2 = System.nanoTime();
            System.out.println(i * 10000 + " " + isMatch + " " + (time2 - time1)/1e9);
        }
    }
}
